pkg/networkserver/grpc_gsns.go

```diff
@@ -837,26 +836,33 @@ var handleDataUplinkGetPaths = [...]string{
 	"supports_join",
 }
 
+// mergeMetadata merges the metadata collected for up.
+// mergeMetadata mutates up.RxMetadata discarding any existing up.RxMetadata value.
+// NOTE: Since events are published async we need ensure that up passed to an event earlier is not mutated,
+// hence up is taken by value here.
+func (ns *NetworkServer) mergeMetadata(ctx context.Context, up *ttnpb.UplinkMessage) {
```

Comment:
Taken by value? It looks like it's passed copied

Reply:
I had a different implementation originally